airflow/settings.py

```diff
@@ -51,6 +52,38 @@
 log.info("Configured default timezone %s" % TIMEZONE)
 
 
+class RetryingQuery(Query):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        try:
+            self.max_tries = max(1, conf.getint('core', 'SQL_ALCHEMY_STATEMENT_MAX_TRIES'))
+        except conf.AirflowConfigException:
+            self.max_tries = 1
+
+        try:
+            self.max_retry_time_seconds = max(0, conf.getint('core', 'SQL_ALCHEMY_STATEMENT_MAX_RETRY_SECONDS'))
+        except conf.AirflowConfigException:
+            self.max_retry_time_seconds = 30
+
+    def __iter__(self):
+        try_number = 1
+        last_exc = None
+        while try_number <= self.max_tries:
+            try:
+                return super().__iter__()
+
+            except Exception as ex:
+                log.warning(f'Try {try_number}/{self.max_tries} failed to perform db action.', exc_info=ex)
+                last_exc = ex
+                time.sleep(min(float(self.max_retry_time_seconds), 0.1 * (1 << try_number)))
+
+            try_number += 1
+
+        raise Exception(f'Failed to perform db action after {self.max_tries} attempts.') from last_exc
```

Comment:
I don't think we should wrap this. I don't want to potentially break existing queries that handle, or need the actual exception that postgres returns.

Reply:
hmm good point; i can just `raise last_exc`